lib/src/query.c

```diff
@@ -452,6 +458,296 @@ static void capture_list_pool_release(CaptureListPool *self, uint16_t id) {
   self->free_capture_list_count++;
 }
 
+/**************
+ * Quantifiers
+ **************/
+
+static TSQuantifier quantifier_mul(
+  TSQuantifier left,
+  TSQuantifier right
+) {
+  TSQuantifier result = Zero; // initialized to make compiler happy, but all cases should be covered below!
+  switch (left)
+  {
+    case Zero:
+      result = Zero;
```

Comment:
That is what I had at first, but the compiler kept complaining about a missing default case. Since there is no meaningful default, I wrote it like this, so the uninitialized state acts as a default. This default is never returned (bar invalid inputs) since the case is exhaustive. However, the compiler (in CI at least) wouldn't realize that 🤷🏻.

If you prefer early returns, I can write it as such, use an arbitrary default value and add a big comment explaining that the default value is really meaningless.

Reply:
Actually, after rereading the code, I see that the result is not even uninitialized. I'll compact this a bit, then :+1: